#! /bin/sh -e

# DP: Fix email/Message.py set_charset() barfing on unicode

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Lib/email/Message.py~	2008-03-07 02:43:43.000000000 +0000
+++ Lib/email/Message.py	2007-03-09 15:06:01.000000000 +0000
@@ -238,6 +238,8 @@
             return
         if isinstance(charset, str):
             charset = Charset.Charset(charset)
+        if isinstance(charset, unicode):
+            charset = Charset.Charset(str(charset))
         if not isinstance(charset, Charset.Charset):
             raise TypeError(charset)
         # BAW: should we accept strings that can serve as arguments to the
